tools/CodeGen/ExceptionThrowerGenerator.cs

```diff
@@ -0,0 +1,109 @@
+namespace CodeGen;
+
+using Microsoft.CodeAnalysis;
+
+[Generator]
+public class ExceptionThrowerGenerator : StronglyTypedOverloadsGenerator
+{
+    public override void Execute(GeneratorExecutionContext context)
+    {
+        var source = $$"""
+            //------------------------------------------------------------------------------
+            // <auto-generated>
+            //     This code was generated by {{this.GetType().FullName}}.
+            //
+            //     Changes to this file will be lost when the project is rebuilt.
+            // </auto-generated>
+            //------------------------------------------------------------------------------
+            namespace FakeItEasy.Configuration
+            {
+                using System;
+                using System.Reflection;
+                using FakeItEasy.Core;
+
+                public partial interface IExceptionThrowerConfiguration<out TInterface>
+                {
+            {{Indent(2, GenerateMethodDeclarations())}}
+                }
+
+                internal partial class RuleBuilder
+                {
+            {{Indent(2, GenerateMethodImplementations("IVoidConfiguration"))}}
+
+                    public partial class ReturnValueConfiguration<TMember>
+                    {
+            {{Indent(3, GenerateMethodImplementations("IReturnValueConfiguration<TMember>"))}}
+                    }
+                }
+
+                internal partial class AnyCallConfiguration
+                {
+            {{Indent(2, GenerateMethodImplementations("IVoidConfiguration"))}}
+                }
+
+                internal partial class PropertySetterConfiguration<TValue>
+                {
+            {{Indent(2, GenerateMethodImplementations("IPropertySetterConfiguration"))}}
+
+                    private partial class PropertySetterAdapter
+                    {
+            {{Indent(3, GenerateMethodImplementations("IPropertySetterConfiguration"))}}
+                    }
+
+                    private partial class PropertySetterAfterCallbackConfiguredAdapter
+                    {
+            {{Indent(3, GenerateMethodImplementations("IPropertySetterConfiguration"))}}
+                    }
+                }
+            }
+            """;
+        context.AddSource("ExceptionThrower.StronglyTyped.cs", source);

```

Comment:
You were worried about having long file names. We could shave a few chars by changing ".StronglyTyped.cs" to ".g.cs" (which seems to be the convention used by .NET for generated files)

Reply:
191 is the max length now, even with the FakeItEasy.Extensions.ValueTask generated files.